### PR TITLE
[PVR] Guide window: fix ruler date might taking to much space (result ing in ugly overlap).

### PR DESCRIPTION
**Pull request by ksooo**
Before:

screenshot002

After:

screenshot000

@phil65 fyi
@Jalle19 mind taking a look at the code changes

### CODE REVIEW

**Review by Jalle19 — APPROVED** [review]
Looks good to me

**Comment by ksooo** [comment]
Jenkins build this please

File: xbmc/XBDateTime.cpp [hunk 1]
```diff
@@ -1327,12 +1327,12 @@ std::string CDateTime::GetAsLocalizedTime(const std::string &format, bool withSe
...
 
-std::string CDateTime::GetAsLocalizedDate(bool longDate/*=false*/) const
+std::string CDateTime::GetAsLocalizedDate(bool longDate/*=false*/, bool withYear /* = true */) const
 {
-  return GetAsLocalizedDate(g_langInfo.GetDateFormat(longDate));
+  return GetAsLocalizedDate(g_langInfo.GetDateFormat(longDate), withYear);
 }
 
-std::string CDateTime::GetAsLocalizedDate(const std::string &strFormat) const
+std::string CDateTime::GetAsLocalizedDate(const std::string &strFormat, bool withYear /* = true */) const
```
Comment (xhaggi) [inline xbmc/XBDateTime.cpp]: why do we need this here? you should be able to leave the year out of the format pattern.

**Comment by xhaggi** [comment]
I'm not very familiar with our date time formatting class, but why not use a pattern for the date we need in the guide ruler for example like java does with its date time pattern `E, d M`.

**Comment by xhaggi** [comment]
after a second look it should be possible right now `DDD,  D MMM` or `ddd, d mmm`. Both do the same because it's case-insensitve.

**Comment by xhaggi** [comment]
sorry for the noise again but IMO this should be handled by the skin because other skins don't have an issue with limited space for that label e.g. Xperience1080.

-1 from my side

**Comment by ksooo** [comment]
Regarding format string: you cannot hardcode one because it's depended on the locale. Thus, my approach of obtaining the correct locale depended format string and than to remove the year is correct and definitely better than hardcoding something locale independent like you suggested. It is fine the same way for GetLocalizedTime(..., bool withoutSeconds), btw

Regarding coding it in the skin: @phil65 told me that this is currently not possible.

File: xbmc/XBDateTime.cpp (hunk 1 above)
Comment (ksooo) [inline xbmc/XBDateTime.cpp]: true, but i did it exactly like it was done for <code>GetAsLocalizedTime</code>. i wanted to change as less code as possible at this stage in the release cycle.

Comment (ksooo) [inline xbmc/XBDateTime.cpp]: See line line 1332. There it is called.

Comment (Paxxi) [inline xbmc/XBDateTime.cpp]: I have a real dislike for bool parameters, I'd rather see a new method like GetAsLocalizedDateWithoutYear as it makes it very clear what it does without looking at the declaration

Comment (ksooo) [inline xbmc/XBDateTime.cpp]: > GetAsLocalizedDateWithoutYear

to get rid of the two bools we would end with GetAsLongLocalizedDateWithoutYear for the guide window ruler case. fine by me.

**Comment by ksooo** [comment]
On second thought, I'm with you @xhaggi. This should be solved in the skin. If this is currently not possible this should be tackled by extending guiinfo, not by hardcoding the (year-less) date format in the core.

**Comment by ksooo** [comment]
@phil65 fyi, closing this PR.

**Comment by xhaggi** [comment]
@ksooo You are right, if the predefined format pattern is locale depended. Adding more localized pattern seems reasonable for me as doing it like this.

Anyway, I don't want to miss the year in the guide window only because our default skin is not able to handle it right. At this stage we should fix it there. Modifying the core will effects all skins.

**Comment by xhaggi** [comment]
wow .. simultaneously 😄

**Pull request closed without merging**